paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/CreateLinkState.kt

```diff
@@ -0,0 +1,289 @@
+package com.stripe.android.paymentsheet.state
+
+import android.os.Parcelable
+import com.stripe.android.DefaultCardBrandFilter
+import com.stripe.android.common.model.CommonConfiguration
+import com.stripe.android.link.LinkConfiguration
+import com.stripe.android.link.account.LinkStore
+import com.stripe.android.link.gate.LinkGate
+import com.stripe.android.link.model.AccountStatus
+import com.stripe.android.link.model.toLoginState
+import com.stripe.android.link.ui.inline.LinkSignupMode
+import com.stripe.android.lpmfoundations.luxe.isSaveForFutureUseValueChangeable
+import com.stripe.android.lpmfoundations.paymentmethod.PaymentSheetCardBrandFilter
+import com.stripe.android.lpmfoundations.paymentmethod.toPaymentSheetSaveConsentBehavior
+import com.stripe.android.model.ElementsSession
+import com.stripe.android.model.ElementsSession.Flag.ELEMENTS_MOBILE_FORCE_SETUP_FUTURE_USE_BEHAVIOR_AND_NEW_MANDATE_TEXT
+import com.stripe.android.model.LinkDisabledReason
+import com.stripe.android.model.LinkSignupDisabledReason
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.paymentelement.confirmation.utils.sellerBusinessName
+import com.stripe.android.payments.financialconnections.GetFinancialConnectionsAvailability
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.repositories.CustomerRepository
+import kotlinx.parcelize.Parcelize
+import javax.inject.Inject
+
+internal interface CreateLinkState {
```

Comment:
👀 This is an interactor extracted out of `DefaultPaymentElementLoader`, then refactored to incorporate disabled reasons.